subsys/bluetooth/host/gatt.c

```diff
@@ -1675,6 +1680,13 @@ int bt_gatt_service_register(struct bt_gatt_service *svc)
 	__ASSERT(svc->attrs, "invalid parameters\n");
 	__ASSERT(svc->attr_count, "invalid parameters\n");
 
+	if (IS_ENABLED(CONFIG_BT_SETTINGS) &&
+	    atomic_get(&init) &&
```

Comment:
Not exactly related to this PR, but it had completely flown past me that we have these rather unintuitive `init` and `service_init` variables in `gatt.c`. It seems they're essentially used as atomic boolean flags to track whether initialisation has been done or not. I think it make much more sense to convert both of them into a single atomic_t bitmask declared using `ATOMIC_DEFINE()`, and then instead of the `cas()` or `atomic_get()` calls, you'd have something like `if (atomic_test_and_set_bit(gatt_flags, GATT_INIT_DONE)) {` in `gatt_init()` and here above you'd have `atomic_test_bit(gatt_flags, GATT_INIT_DONE)` instead of the awkward `atomic_get(&init)`.

Reply:
yeah I also found that weird, I'll address it.